dashboard/test/models/ability_test.rb

```diff
@@ -144,6 +151,13 @@ class AbilityTest < ActiveSupport::TestCase
     assert ability.can?(:manage, FrequentUnsuccessfulLevelSource)
   end
 
+  test 'teachers read Section' do
+    ability = Ability.new(create(:teacher))
+
+    assert ability.can?(:read, Section)
+    assert ability.can?(:manage, Section)
+  end
```

Comment:
See https://github.com/ryanb/cancan/wiki/checking-abilities#checking-with-class. Checking access with the class (rather than an instance) ignores the conditions and will always return true (for use in Index). So we should test specific instances too.